lang/src/org/partiql/lang/eval/physical/window/Lead.kt

```diff
@@ -0,0 +1,41 @@
+package org.partiql.lang.eval.physical.window
```

Comment:
The diff between `lag` and `lead` implementation appears to be minimal:
```bash
10,12c10,11
< internal class Lead : NavigationWindowFunction() {
<
<     override val name = "lead"
---
> internal class Lag : NavigationWindowFunction() {
>     override val name = "lag"
29a29
>
31c31
<         val targetIndex = currentPos + offsetValue
---
>         val targetIndex = currentPos - offsetValue
33c33
<         if (targetIndex <= currentPartition.lastIndex) {
---
>         if (targetIndex >= 0 && targetIndex <= currentPartition.lastIndex) {
```

does it make sense to reduce the duplicate code further?

Reply:
I tried to come up with a `LagLeadCommon`. It is entirely possible, but later decided to cut the functions into two separate classes, so it is a little bit more obvious to see the logic here. We can decide whether we shall do the refactor after we implemented other window functions. May be a todo note here?

Reply:
Having a `TODO` makes sense.